python/tvm/tir/schedule/schedule.py

```diff
@@ -63,7 +63,37 @@ def __init__(self) -> None:
 RAND_VAR_TYPE = Union[ExprRV, BlockRV, LoopRV]  # pylint: disable=invalid-name
 
 # Update to `Literal["detail", "fast", "none"]` once upgraded to python3.8
-ERROR_RENDER_LEVEL_CANDIDATES = Union[str]  # pylint: disable=invalid-name
+_ERROR_RENDER_LEVEL = {
+    "detail": 0,
+    "fast": 1,
+    "none": 2,
+}
+
+
+def _preprocess_constructor_arguments(
+    mod: Union[PrimFunc, IRModule],
+    debug_mode: Union[bool, int] = False,
+    error_render_level: str = "detail",
+) -> Tuple[IRModule, int, int]:
+    # preprocess `mod`
+    if isinstance(mod, PrimFunc):
+        mod = IRModule({"main": mod})
+    # preprocess `debug_mode`
+    if isinstance(debug_mode, bool):
+        if debug_mode:
+            debug_mode = -1
```

Comment:
Are you assuming two-complement integer representation or unsigned underflow here? I'd prefer that the mask was set more explicitly.

Reply:
I don’t actually like unsigned stuff. In the backend, schedule explicitly does the following check:
- if the mask is -1, then all debug flags are on
- if the mask is non-negative, do bit manipulation to figure out which debug flags are on
- the mask is not allowed to be any negative number other than -1

Reply:
Its just a little confusing because debug_mode can be -1, some non-negative integer, true, or false. Not sure about a better solution. Maybe just remove the true/false?

Reply:
True and False are actually the two most commonly used options - turn it on or turn it off, so I feel like it reasonable to make such convenience for our users :-)

Reply:
Another possible alternative would be provide EnumInt.

e.g. `debug_mask=tir.DebugMask.ALL `

Reply:
To be more pythonic, I would prefer to use “on”/“off”/true/false as user-facing interface, and do the type conversion internally

Reply:
Is this code ever going to be ported to C++? In that case, I think trying to unify `debug_mode` into one type would be good for ease of conversion in the future.

Reply:
My understanding is that this is a pythonic wrapper to makes it friendly to python ecosystem, while c++ side will have its own set of c++ focused API

Reply:
Yeah the actual code is completely written in C++, and it is strongly typed with only one type. The dynamically-typed syntactic sugar here is to just make it more "pythonic". Therefore there is no actual logic here so we don't need to worry at all :-)

BTW, it is super important that we write all the main logic in C++ if possible, and the TensorIR/meta schedule project are implementing this rule :-)

Reply:
Let us use str option then which will be more explicit, e.g. `"all"` to indicate all options and `"none"` to indicate none of the options are selected

Reply:
I still think we should stick with `True` and `False`. If we go with "all" and "none", it is cmake-style, and the problem is that it's hard to remember. User may get confused: should I use "on" and "off", "all" and "none", could I use None, etc. It only makes thing complicated.